src/error-handling.js

```diff
@@ -0,0 +1,80 @@
+const path = require('path');
+const fs = require('fs-extra');
+
+// Directory-name must exist, and Directory-name must point to valid directory
+async function validatePathName (dir) {
+  try {
+    const files = await fs.readdir(dir);
+    return files;
+  } catch (err) {
+    if (err.name.startsWith('TypeError')) {
+      err.subMessage = `Must include a directory name in the command 'convert-json-schemas-to-graphql-types <directory-name>'`;
+    }
+
+    if (err.errno === -2) {
+      err.subMessage = `The path name "${err.path}" is not a valid directory`;
+    }
+
+    throw err;
+  }
+}
+
+// Each file must have .json extension, and each file must be syntactically correct, and no file is an array of schema
+async function validateJSONSyntax (file, dir) {
+  if (path.extname(file) !== '.json') {
+    const err = new TypeError(`All files in directory must have .json extension`);
+    err.subLocation = dir.endsWith('/') ? `${dir}${file}` : `${dir}/${file}`;
+    throw err;
+  }
+
+  try {
+    const fileContent = await fs.readFile(path.join(dir, file));
+    const parsedFileContent = JSON.parse(fileContent);
+    if (JSON.stringify(parsedFileContent).startsWith('[')) {
```

Comment:
Instead of parsing and the stringifying, I think you could use `Array.isArray(parsedFileContext)` instead.

Also, I think the TypeError message should be changed to be focused on the value of the file and not the syntax, I like the subMessage much better. 

I'm ok with adding this restriction now, but perhaps after this PR is merged, you could add support for array files too? I would not tackle that as part of this PR.

Reply:
I created issue #20 to track support for arrays